### PR TITLE
Feature/commands

### PR DESCRIPTION
**Pull request by Sefiraat**
Adds ACF and a few basic commands to manage mobcaps including a permission node tree

### CODE REVIEW

File: src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        if (sender instanceof Player player) {
+            for (String message : messages) {
+                player.sendMessage(Theme.CLICK_INFO.apply(message));
+            }
+        } else {
+            for (String message : messages) {
+                Netheopoiesis.logInfo(message);
+            }
```
Comment (J3fftw1) [inline src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java:48]: we are looping through the for in either cases why dont we have it outside the if?

File: src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Subcommand("PurgeMobCap")
+    @Description("Kills all mobs from a specific Mob Cap")
+    @CommandPermission("netheopoiesis.admin.mobcaps")
+    @CommandCompletion("@MOB_CAPS")
+    public void purgeMobCap(CommandSender sender, String mobCapType) {
+        final MobCap cap = MobManager.getInstance().getMobCap(MobCapType.valueOf(mobCapType));
+        final String message = "Mob Cap Purged";
+
+        cap.killAllMobs();
```
Comment (J3fftw1) [inline src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java:60]:
```suggestion
        final MobCap mobCap = MobManager.getInstance().getMobCap(MobCapType.valueOf(mobCapType));
        final String message = "Mob Cap Purged";

       mobCap.killAllMobs();
```

File: src/main/resources/plugin.yml [hunk 3]
```diff
@@ -10,4 +10,18 @@ depend:
...
+    children:
+      netheopoiesis.admin: true
+  netheopoiesis.admin:
+    description: Provides access to all Netheo admin functions
+    default: op
+    children:
+      netheopoiesis.admin.mobcaps: true
+  netheopoiesis.admin.mobcaps:
+    description: Allows the user to see and manage Nethero MobCaps
+    default: op
```
Comment (J3fftw1) [inline src/main/resources/plugin.yml:27]: does plugin.yml properly update?

File: src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        registerCommand(new NetheoCommands());
+
+        getCommandCompletions().registerCompletion(
+            "MOB_CAPS",
+            context -> Arrays.stream(MobCapType.values())
+                             .map(Enum::name)
+                             .collect(Collectors.toSet())
+        );
+    }
```
Comment (J3fftw1) [inline src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java:30]: Is this formatted properly it looks a little weird

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/resources/plugin.yml (hunk 3 above)
Comment (Sefiraat) [reply src/main/resources/plugin.yml:27]: How do you mean?

File: src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java (hunk 4 above)
Comment (Sefiraat) [reply src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java:30]: Yeah, steams look weird, especially if in another method. I could expand the lamba if you like, but I think this is fine?

File: src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java (hunk 1 above)
Comment (Sefiraat) [reply src/main/java/dev/sefiraat/netheopoiesis/implementation/commands/NetheoCommands.java:48]: Ha yeah much better xD

File: pom.xml [hunk 5]
```diff
@@ -152,9 +156,14 @@
...
             <artifactId>annotations</artifactId>
-            <version>RELEASE</version>
+            <version>23.0.0</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>co.aikar</groupId>
+            <artifactId>acf-paper</artifactId>
+            <version>0.5.1-SNAPSHOT</version>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (Sefiraat) [reply pom.xml:166]: image

A bit like Paper lib I assume

File: src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java (hunk 4 above)
Comment (J3fftw1) [reply src/main/java/dev/sefiraat/netheopoiesis/managers/DispatchManager.java:30]: nah its fine

File: src/main/resources/plugin.yml (hunk 3 above)
Comment (J3fftw1) [reply src/main/resources/plugin.yml:27]: remember the convo we had on discord about updating yml

Comment (Sefiraat) [reply src/main/resources/plugin.yml:27]: Oh, that doesn’t apply here, plugin.yml doesn’t leave the jar

**Review by J3fftw1 — APPROVED** [review]
Lgtm

**Pull request merged**